src/plugins/search/ngv-search-providers.ts

```diff
@@ -0,0 +1,23 @@
+import type {INGVSearchProvider} from '../../interfaces/search/ingv-search-provider.js';
+
+const factories: Record<string, () => Promise<{provider: INGVSearchProvider}>> =
+  {
+    // Here we register singletons for builtin providers
+    geoadmin: () => import('./ngv-geoadmin-provider.js'),
+    // add custom singletons here or through the setter
+  };
+
+export function setProviderFunction(
+  key: string,
+  fn: () => Promise<{provider: INGVSearchProvider}>,
+): void {
+  factories[key] = fn;
+}
```

Comment:
This is redundant with the "add custom singletons" above.
The pros: it would appear in an (automated) documentation.